Connectors/Rhino/Speckle.Connectors.RhinoShared/HostApp/Properties/PropertiesExtractor.cs

```diff
@@ -0,0 +1,36 @@
+using Rhino.DocObjects;
+
+namespace Speckle.Connectors.Rhino.HostApp.Properties;
+
+/// <summary>
+/// Extracts properties for rhino objects.
+/// </summary>
+public class PropertiesExtractor
+{
+  private readonly UserStringsExtractor _userStringsExtractor;
+
+  public PropertiesExtractor(UserStringsExtractor userStringsExtractor)
+  {
+    _userStringsExtractor = userStringsExtractor;
+  }
+
+  public Dictionary<string, object?> GetProperties(RhinoObject rhObject)
+  {
+    Dictionary<string, object?> properties = new();
+    AddDictionaryToPropertyDictionary(_userStringsExtractor.GetUserStrings(rhObject), "User Strings", properties);
+
+    return properties;
+  }
+
+  private void AddDictionaryToPropertyDictionary(
+    Dictionary<string, object?>? entryDictionary,
+    string entryName,
+    Dictionary<string, object?> propertyDictionary
+  )
+  {
```

Comment:
i do not believe we need this function at all, i would just put if logic to proper place. Also, It says `Add...To...` but it doesnt guarntee it will add since it has if statement. At least i would rename if to `Try...Add...To`